src/EditorFeatures/VisualBasicTest/Diagnostics/MakeMethodSynchronous/MakeMethodSynchronousTests.vb

```diff
@@ -0,0 +1,237 @@
+
+Class C
+    Async Sub [|FooAsync|]()
+    End Sub
+End Class",
+"Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+    End Sub
+End Class",
+compareTokens:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)>
+        Public Async Function TestRenameMethod1() As Task
+            Await TestAsync(
+"Imports System.Threading.Tasks
+
+Class C
+    Async Sub [|FooAsync|]()
+    End Sub
+
+    Sub Bar()
+        FooAsync()
+    End Sub
+End Class",
+"Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+    End Sub
+
+    Sub Bar()
+        Foo()
+    End Sub
+End Class",
+compareTokens:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)>
+        Public Async Function TestSingleLineSubLambda() As Task
+            Await TestAsync(
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Action(of Task) =
+            Async [|Sub|]() Return
+    End Sub
+End Class",
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Action(of Task) =
+            Sub() Return
+    End Sub
+End Class",
+compareTokens:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)>
+        Public Async Function TestSingleLineFunctionLambda() As Task
+            Await TestAsync(
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Func(of Task) =
+            Async [|Function|]() 1
+    End Sub
+End Class",
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Func(of Task) =
+            Function() 1
+    End Sub
+End Class",
+compareTokens:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)>
+        Public Async Function TestMultiLineSubLambda() As Task
+            Await TestAsync(
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Action(of Task) =
+            Async [|Sub|]()
+                Return
+            End Sub
+    End Sub
+End Class",
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Action(of Task) =
+            Sub()
+                Return
+            End Sub
+    End Sub
+End Class",
+compareTokens:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)>
+        Public Async Function TestMultiLineFunctionLambda() As Task
+            Await TestAsync(
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Func(of Task) =
+            Async [|Function|]()
+                Return 1
+            End Function
+    End Sub
+End Class",
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Sub Foo()
+        dim f as Func(of Task) =
```

Comment:
(and if the answer is no: a comment on the test would be appreciated explaining why this test is testing or not testing what it is.)